components/fastly_backends/main.tf

```diff
@@ -1,11 +1,17 @@
-variable "northstar_name" {}
-variable "northstar_domain" {}
-variable "northstar_backend" {}
-variable "rogue_name" {}
-variable "rogue_domain" {}
-variable "rogue_backend" {}
-variable "papertrail_destination" {}
-variable "papertrail_log_format" {}
+variable "name" {
+  description = "The name for this Fastly property."
+  type        = string
+}
+
+variable "backends" {
+  description = "The domains/backends to route traffic to in this property."
+  type        = list(object({ name = string, domain = string, backend = string }))
+}
```

Comment:
This allows us to pass "backend" modules as long as they have `name`, `domain`, and `backend` output variables! This way we don't need to define individual `northstar_name`, `northstar_domain`, `northstar_backend`, `rogue_name`, etc. but can just pass in a list of backends like `module.northstar, module.rogue`.